### PR TITLE
[TOOLS-2894] Scope chart implemented with Highcharts

### PR DESCRIPTION
**Pull request by emannuelalmeida**

### CODE REVIEW

File: core/src/main/java/objective/taskboard/jira/ProjectService.java [hunk 1]
```diff
@@ -185,6 +185,17 @@ public Version getVersion(String versionId) {
                 .collect(toMap(ProjectFilterConfiguration::getProjectKey, p -> p));
     }
 
+    private ProjectFilterConfiguration mergeJiraData(ProjectFilterConfiguration project) {
```
Comment (objective-solutions-user) [inline core/src/main/java/objective/taskboard/jira/ProjectService.java]: MAJOR Remove this unused private "mergeJiraData" method. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AUnusedPrivateMethod)

File: strategical-dashboard/src/main/java/objective/taskboard/monitor/ScopeScenarioService.java [hunk 2]
```diff
@@ -28,18 +29,20 @@
...
 import objective.taskboard.utils.Clock;
 
 @Service
 public class ScopeScenarioService {
 
     private FollowUpSnapshotService followUpSnapshotService;
-    private ScenarioFacade scenarioFacade;
+    private ScenarioComponent scenarioComponent;
     private Clock clock;
+    private final long MAX_WEEKS_FACTOR = 3;
```
Comment (objective-solutions-user) [inline strategical-dashboard/src/main/java/objective/taskboard/monitor/ScopeScenarioService.java]: MINOR Make this final field static too. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS1170)
MINOR Rename this field "MAX_WEEKS_FACTOR" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS00116)

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: strategical-dashboard/src/main/java/objective/taskboard/monitor/ScenarioChartController.java [hunk 3]
```diff
@@ -39,7 +45,7 @@ public ScenarioChartController(ScopeScenarioService scopeScenarioService,
     }
 
     @Cacheable(key = "#projectKey + #zoneId")
-    @GetMapping("projects/{project}/scope")
+    @GetMapping("projects/{project}/chart")
     public ResponseEntity<Object> scope(
```
Comment (taksan) [inline strategical-dashboard/src/main/java/objective/taskboard/monitor/ScenarioChartController.java]: Should change name to projectChartData or something like that

File: strategical-dashboard/web-app/src/app/dashboard/detail/chart/chart.service.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    public conservativeScopeSeries: ChartDataPoint[];
+    public optimisticScopeSeries: ChartDataPoint[];
+    public topOnTrackScopeSeries: ChartDataPoint[];
+    public bottomOnTrackScopeSeries: ChartDataPoint[];
+    public topAttentionScopeSeries: ChartDataPoint[];
+    public bottomAttentionScopeSeries: ChartDataPoint[];
+}
+
+export class ChartDataPoint {
+    public date: string;
```
Comment (taksan) [inline strategical-dashboard/web-app/src/app/dashboard/detail/chart/chart.service.ts]: If date was an actual date, this probably would work better

**Pull request merged**